tools/perf/util/cs-etm.c

```diff
@@ -809,8 +911,13 @@ static int cs_etm__sample(struct cs_etm_queue *etmq)
 
         err = cs_etm_decoder__get_packet(etmq->decoder,&packet);
         // if there is no sample, it returns err = -1, no real error
-
-        if (!err && packet.sample_type & CS_ETM_RANGE) {
+        if (err)
+                return err;
+        if (etmq->etm->synth_opts.last_branch &&
+            etmq->last_branch_rb->nr < etmq->etm->synth_opts.last_branch_sz) {
+                // FIXME: need to synth partial stack on last event.
```

Comment:
I this work you intend to do in this patchset?  What is currently missing that prevents you from implementing?